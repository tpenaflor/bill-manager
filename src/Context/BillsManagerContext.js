import React from 'react'

const BillsManagerContext = React.createContext()

export default BillsManagerContext