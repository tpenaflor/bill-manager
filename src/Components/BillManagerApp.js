import React, {useEffect, useReducer} from 'react'

import AddBill from './AddBill.js'
import AddParticipant from './AddParticipant.js'
import BillList from './BillList.js'
import BillSummary from './BillSummary.js'
import Participants from './Participants.js'

import billReducer from '../Reducer/BillReducer'
import participantReducer from '../Reducer/ParticipantReducer'
import transactionReducer from '../Reducer/TransactionReducer'

import BillsManagerContext from '../Context/BillsManagerContext'

const BillManagerApp = () => {
  const [bills, billDispatch] = useReducer(billReducer, [])
  const [participants, partDispatch] = useReducer(participantReducer, [])
  const [transactions, transDispatch] = useReducer(transactionReducer, [])

  useEffect(() => {
      const localBills = JSON.parse(localStorage.getItem('bills'))
      const localParicipant = JSON.parse(localStorage.getItem('participants'))
      const localTransactions = JSON.parse(localStorage.getItem('transactions'))

      if (localBills){
        billDispatch({type:"POPULATE", bills:localBills})
      }
      if (localParicipant){
        partDispatch({type:"POPULATE", participants:localParicipant})
      }
      if (localTransactions){
        transDispatch({type:"POPULATE", transactions:localTransactions})
      }
  },[])

  useEffect(() =>{
      localStorage.setItem('bills', JSON.stringify(bills))
    },[bills])

  useEffect(() =>{
    localStorage.setItem('participants', JSON.stringify(participants))
  },[participants])

  useEffect(() =>{
      localStorage.setItem('transactions', JSON.stringify(transactions))
    },[transactions])

  return (
    <BillsManagerContext.Provider value={{bills, participants, transactions, transDispatch, billDispatch, partDispatch}}>
      <h1>Bill Manager</h1>
      <AddBill />
      <AddParticipant />
      <Participants />
      <BillList/>
      <BillSummary />
    </BillsManagerContext.Provider>    
  )
}

export default BillManagerApp