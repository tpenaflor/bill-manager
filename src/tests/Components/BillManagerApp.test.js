import React from 'react'
import {shallow} from 'enzyme'
import { shallowToJson  } from 'enzyme-to-json';

import BillManagerApp from '../../Components/BillManagerApp'
import * as BillsManagerContext from '../../Context/BillsManagerContext'
import {billFixture} from '../Fixtures/BillFixture'

test('render bill list', ()=> {
    const contextValues = {
        bills : billFixture
    }

    jest.spyOn(BillsManagerContext, 'useBillsManagerContext')
        .mockImplementation(() => contextValues)

    const wrapper = shallow(<BillManagerApp />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
})