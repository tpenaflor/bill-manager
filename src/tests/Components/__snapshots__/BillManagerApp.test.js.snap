// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render bill list 1`] = `
<ContextProvider
  value={
    Object {
      "billDispatch": [Function],
      "bills": Array [],
      "partDispatch": [Function],
      "participants": Array [],
    }
  }
>
  <h1>
    Bill Manager
  </h1>
  <AddBill />
  <AddParticipant />
  <Participants />
  <BillList />
  <BillSummary />
</ContextProvider>
`;
